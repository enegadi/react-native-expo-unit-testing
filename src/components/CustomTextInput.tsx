import React from "react";
import { TextInput, StyleSheet } from "react-native";

interface CustomTextInputProps {
  placeholder: string;
  onChangeText: (text: string) => void;
  value: string;
}

const CustomTextInput = (props: CustomTextInputProps) => {
  const { placeholder, onChangeText, value } = props;

  return (
    <TextInput
      style={styles.input}
      placeholder={placeholder}
      onChangeText={onChangeText}
      value={value}
    />
  );
};

const styles = StyleSheet.create({
  input: {
    height: 40,
    borderColor: "blue",
    borderWidth: 2,
    borderRadius: 10,
    paddingLeft: 10,
    paddingRight: 10,
    fontSize: 16,
    color: "black",
    backgroundColor: "white",
    shadowColor: "gray",
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.3,
    shadowRadius: 4,
    elevation: 3,
  },
});

export default CustomTextInput;
