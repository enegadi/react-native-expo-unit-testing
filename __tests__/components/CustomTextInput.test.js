import React from "react";
import renderer from "react-test-renderer";
import CustomTextInput from "../../src/components/CustomTextInput";

describe("<CustomTextInput />", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<CustomTextInput />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("has 1 child", () => {
    const tree = renderer.create(<CustomTextInput />).toJSON();
    expect(tree.children.length).toBe(1);
  });

  it("has a specific child component", () => {
    const tree = renderer.create(<CustomTextInput />).toJSON();
    expect(tree.children[0].type).toBe("Text");
  });

  it("has a specific prop value", () => {
    const tree = renderer.create(<CustomTextInput prop1="value1" />).toJSON();
    expect(tree.props.prop1).toBe("value1");
  });

  it("has a default placeholder value", () => {
    const tree = renderer.create(<CustomTextInput />).toJSON();
    expect(tree.props.placeholder).toBe("Enter text");
  });

  it("triggers onChange event", () => {
    const mockOnChange = jest.fn();
    const tree = renderer
      .create(<CustomTextInput onChange={mockOnChange} />)
      .toJSON();
    tree.props.onChangeText("Hello");
    expect(mockOnChange).toHaveBeenCalledWith("Hello");
  });

  it("has a specific style applied", () => {
    const tree = renderer
      .create(<CustomTextInput style={{ color: "red" }} />)
      .toJSON();
    expect(tree.props.style).toEqual({ color: "red" });
  });
});
