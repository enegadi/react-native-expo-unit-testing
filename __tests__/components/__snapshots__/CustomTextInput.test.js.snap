// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CustomTextInput /> renders correctly 1`] = `
<TextInput
  style={
    {
      "backgroundColor": "white",
      "borderColor": "blue",
      "borderRadius": 10,
      "borderWidth": 2,
      "color": "black",
      "elevation": 3,
      "fontSize": 16,
      "height": 40,
      "paddingLeft": 10,
      "paddingRight": 10,
      "shadowColor": "gray",
      "shadowOffset": {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.3,
      "shadowRadius": 4,
    }
  }
/>
`;
